### DISCUSSION TITLE
DMV for Calculated Table Expression

### DISCUSSION THREAD

**Discussion by lisawid** [DaxStudio/DaxStudio#1261] [General]
Hello, I have been looking for hours for a DMV that shows the expression used for a calculated Table, but at this point I have looked through everything and believe no column like this exists anywhere. If it does, please let me know! I would really need this for my model documentation.

**Describe the solution you'd like**
A column "Expression" in the TMSCHEMA_TABLES DMV, like with columns.

**Comment by dgosbell** [comment]
This information is in the TMSCHEMA_PARTITIONS DMV. Rows with a Type=2 are the partitions for Calc tables. You can then link these up with the table details in TMSCHEMA_TABLES

To get the expression in the same query as the table details you could use the new INFO functions (if you are using Power BI)

eg
```
DEFINE
    VAR expressions =
        SELECTCOLUMNS (
            FILTER ( INFO.PARTITIONS (), [Type] = 2 ),
            "TableID", [TableID],
            "Expression", [QueryDefinition]
        )
    VAR tables =
        SELECTCOLUMNS ( INFO.TABLES (), "TableID", [Id], "Name", [Name] )

EVALUATE
NATURALLEFTOUTERJOIN ( tables, expressions )
```

**Reply by lisawid** [reply]
Hi Darren, I now realize my oversight - I was working with my report and not the dataset, so I was only shown the one partition that connects to the dataset! Thank you so much for the quick reply! DAX Studio is an amazing help in properly documenting our models to make them understandable for report users.

**Reply by dgosbell** [reply]
Yes, when you create a composite model you then need to be aware of what you connect to with DAX Studio.

**Reply by lisawid** [reply]
Hi again, I thought I'd try my luck with a follow-up question here instead of spamming the discussions with another (probably obvious) question. 

I used DAX Studio to connect to my locally opened PowerBI dataset to create another report for documenting the dataset's contents. In that documentation report, I am simply displaying all of the metadata on tables, columns, measures etc. which I am retrieving with DAX Studio. However, a refresh of this documentation of course only works now if I never close my local file of the dataset and DAX Studio, because once I close one of the two I have to establish a new connection and get a new server address which I need to update in the documentation report for it to stay connected. Is there any smart way of setting up a system in which I can use DAX Studio to retrieve the metadata of my model but have the documentation report be able to refresh on a schedule like a regular PowerBI dataset/report? Or is there simply no potential for setting it up for anything beyond local & manual refreshes?

**Reply by dgosbell** [reply]
If this is just a documentation pbix file connecting to a main reporting dataset then there should be no reason you need to keep DAX Studio open, the connection should just be between the two pbix files. 

If you are deploying your main report to a workspace hosted on a PPU or premium capacity you could use the XMLA endpoint as the "server" to connect your documentation report to the main report. But if you are on a pro workspace you don't have that option and connecting locally is your only option.

**Reply by lisawid** [reply]
Unfortunately not premium user, so that's out of the question. Thanks a lot for iterating! Right now it is a documentation pbix file that pulls in the metadata of the main pbix through the localhost of DAX Studio, while the main pbix file is open in DAX Studio, or am I missing some way to retrieve this metadata directly between the two pbix files?
